storage/filesystem/dotgit/dotgit.go

```diff
@@ -286,26 +290,37 @@ func (d *DotGit) objectPath(h plumbing.Hash) string {
 //More on 'quarantine'/incoming directory here : https://git-scm.com/docs/git-receive-pack
 func (d *DotGit) incomingObjectPath(h plumbing.Hash) string {
 	hString := h.String()
-	directoryContents, err := d.fs.ReadDir(objectsPath)
-	if err != nil {
+
+	if d.incomingDirName == "" {
 		return d.fs.Join(objectsPath, hString[0:2], hString[2:40])
 	}
-	var incomingDirName string
-	for _, file := range directoryContents {
-		if strings.Split(file.Name(), "-")[0] == "incoming" && file.IsDir() {
-			incomingDirName = file.Name()
+
+	return d.fs.Join(objectsPath, d.incomingDirName, hString[0:2], hString[2:40])
+}
+
+// hasIncomingObjects searches for an incoming directory and keeps its name
+// so it doesn't have to be found each time an object is accessed.
+func (d *DotGit) hasIncomingObjects() bool {
+	if !d.incomingChecked {
+		directoryContents, err := d.fs.ReadDir(objectsPath)
+		if err == nil {
+			for _, file := range directoryContents {
+				if strings.Split(file.Name(), "-")[0] == "incoming" && file.IsDir() {
```

Comment:
Is it the same as `strings.HasPrefix(file.Name(), "incoming")`

Reply:
Changed